DALOptimizer/Program.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Windows.Forms;
+using ICSharpCode.NRefactory.CSharp;
+using ICSharpCode.NRefactory.CSharp.Refactoring;
+using ICSharpCode.NRefactory.CSharp.Resolver;
+using ICSharpCode.NRefactory.Editor;
+using ICSharpCode.NRefactory.PatternMatching;
+using ICSharpCode.NRefactory.TypeSystem;
+
+namespace DALOptimizer
+{
+    class Program
+    {
+        [STAThread]
+        public static void Main(string[] args)
+        {
+            OpenFileDialog opnfd = new OpenFileDialog();
+            opnfd.ShowDialog();
+            string fileName = opnfd.FileName;
+            
+            if (string.IsNullOrWhiteSpace(fileName)){
+                MessageBox.Show("Please Select one file ");
+                return;
+            }
+
+            Solution solution = new Solution(fileName);
+            solution.ChooseCSProjFile(fileName);
+
+            MatchInvocation matchInvocation = new MatchInvocation();
+            MatchExpr matchExpr = new MatchExpr();
+
+            // Capture patterns & store it in respective lists of Node patterns
+            foreach (var file in solution.AllFiles)
+            {
+                if (Path.GetFileName(file.FileName) == "DatabaseProcessing.cs")
+                       continue;
+                var astResolver = new CSharpAstResolver(file.Project.Compilation, file.syntaxTree, file.UnresolvedTypeSystemForFile);
+                matchInvocation.CheckAllInvocation(file, astResolver);
+            }
+
+            //PrintFunction pr = new PrintFunction();
+            //pr.PrintMethod(solution);
+
+            Console.Write("Apply refactorings? ");
```

Comment:
@sushantyelpale probably we should suggest here as end user has to type "yes" or "y" to apply refctoring?